src/connected/autobinding.ts

```diff
@@ -156,4 +165,161 @@ export class AutoBindingService {
         return false;
       });
   }
+
+  private async getBestHitsForConnections(connectionToServerProjects: Map<BaseConnection, ServerProject[]>,
+                                          unboundFolder: VSCode.WorkspaceFolder):
+    Promise<Map<BaseConnection, MatchHit[]>> {
+    const folderName = unboundFolder.name;
+    const workspaceName = VSCode.workspace.name;
+    const folderNameTokens = [...tokenizeString(folderName)];
+    const workspaceNameTokens = [...tokenizeString(workspaceName)];
+    const connectionToBestHits = new Map<BaseConnection, MatchHit[]>();
+    for (const [connection, projects] of connectionToServerProjects) {
+      let bestHits: MatchHit[] = [];
+      bestHits.push({ hits: 0, projectKey: '', connection: { connectionId: '' } });
+      for (const project of projects) {
+        const projectKey = project.key;
+        const projectName = project.name;
+        const serverProjectString = (projectKey + projectName).toLowerCase();
+        const folderNameHits = this.getHits(folderNameTokens, serverProjectString);
+        const workspaceNameHits = this.getHits(workspaceNameTokens, serverProjectString);
```

Comment:
Maybe we could optimise this by
- get hits for workspace
- if no hits, then get hits for folder
It could eliminate unnecessary processing

Reply:
I don't think so. Because we need to match by both and choose best one.